### PR TITLE
[fix](array-funcs)fix array funcs which ret can be nullable

### PR DESCRIPTION
**Pull request by amorynan**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
before this pr, we has some array funcs which ret sign is nullable(uint8) , if we got null , we just fill with nullmap but without filling with true/false ret data, then we my has wrong answer like:
```

mysql> SELECT pk, array_contains(col_array_string__undef_signed, col_varchar_1024__undef_signed_index_inverted_p_e) or col_boolean_undef_signed  FROM table_30_un_pa_ke_pr_di3 where  pk = 150;
+------+---------------------------------------------------------------------------------------------------------------------------------+
| pk   | (array_contains(col_array_string__undef_signed, col_varchar_1024__undef_signed_index_inverted_p_e) OR col_boolean_undef_signed) |
+------+---------------------------------------------------------------------------------------------------------------------------------+
|  150 |                                                                                                                             -65 |
+------+---------------------------------------------------------------------------------------------------------------------------------+
```
### Check List (For Committer)

- Test 

    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

File: regression-test/suites/inverted_index_p0/test_array_with_inverted_index_all_type.groovy [hunk 1]
```diff
@@ -267,6 +267,10 @@ suite("test_array_with_inverted_index_all_type"){
                 assertEquals(101, json.NumberLoadedRows)
             }
         }
+
+        // insert into null array
+        sql """insert into ${tableName} (k1) values(101);"""
+        qt_sql_null """ select * from ${tableName} where k1 = 101; """
```
Comment (eldenmoon) [inline regression-test/suites/inverted_index_p0/test_array_with_inverted_index_all_type.groovy:273]: is this query suitable for function_array_xxx ?

Comment (amorynan) [reply regression-test/suites/inverted_index_p0/test_array_with_inverted_index_all_type.groovy:273]: here just insert null data , below we have array_contains query

File: be/src/vec/functions/array/function_array_index.h [hunk 2]
```diff
@@ -217,7 +217,7 @@ class FunctionArrayIndex : public IFunction {
         const auto& right_chars = reinterpret_cast<const ColumnString&>(right_column).get_chars();
 
         // prepare return data
-        auto dst = ColumnVector<ResultType>::create(offsets.size());
+        auto dst = ColumnVector<ResultType>::create(offsets.size(), 0);
```
Comment (airborne12) [inline be/src/vec/functions/array/function_array_index.h:220]: Why filled with 0 with dst column, not dst_null_column?

Comment (amorynan) [reply be/src/vec/functions/array/function_array_index.h:220]: dst can be  nullable(uint8) which is nullable(true/false)

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: be/src/vec/functions/array/function_array_index.h [hunk 3]
```diff
@@ -217,7 +217,7 @@ class FunctionArrayIndex : public IFunction {
...
         auto& dst_data = dst->get_data();
         auto dst_null_column = ColumnUInt8::create(offsets.size());
```
Comment (airborne12) [inline be/src/vec/functions/array/function_array_index.h:222]: need to set default value to 0 like other function.

Comment (amorynan) [reply be/src/vec/functions/array/function_array_index.h]: done!

**Comment by amorynan** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**